core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -224,20 +227,30 @@ public ClientFactoryBuilder connectionPoolListener(
     }
 
     /**
+     * Sets the {@link MeterRegistry} which collects various stats.
+     */
+    public ClientFactoryBuilder meterRegistry(MeterRegistry meterRegistry) {
+        this.meterRegistry = requireNonNull(meterRegistry, "meterRegistry");
+        return this;
+    }
+
+    /**
      * Returns a newly-created {@link ClientFactory} based on the properties of this builder.
      */
     public ClientFactory build() {
+        final MeterRegistry meterRegistry = this.meterRegistry != null ? this.meterRegistry
```

Comment:
Ah good point on `firstNonNull`. For `Optional`, I was just thinking this would become `meterRegistry.orElseGet(SimpleMeterRegistry::new)`. My feeling on `Optional`is we prefer it to nullable types when possible, but their lower performance makes them impractical in some cases. But either way is fine.